apps/src/code-studio/pd/components/useRegionalPartner.jsx

```diff
@@ -57,30 +57,29 @@ export const useRegionalPartner = data => {
   const [partner, setPartner] = useState(null);
   // cache this debounced function so that it is more easily testable
   const debouncedFetch = useCallback(
-    debounce(fetchRegionalPartner, 500, {leading: true}),
+    debounce(
+      data =>
+        fetchRegionalPartner(data)
+          .then(partner => {
+            // Update state with all the partner workshop data to display
+            setLoadingPartner(false);
+            setLoadError(false);
+            // the api returns an object with all fields set to null if not found
+            setPartner(partner.id === null ? null : partner); // TODO(tim): potential issue where requests are fulfilled out of order
```

Comment:
can this be solved by `useEffect` receiving the Promise returned by fetchRegionalPartner(...).then(...), and using that promise to cancel the previous call to useEffect when a subsequent call is made?